internal/oem/oem.go

```diff
@@ -173,3 +204,38 @@ func MustGet(name string) Config {
 func Names() (names []string) {
 	return configs.Names()
 }
+
+func contentsFromString(data string) types.FileContents {
+	return types.FileContents{
+		Source: types.Url{
+			Scheme: "data",
+			Opaque: "," + dataurl.EscapeString(data),
+		},
+	}
+}
+
+func contentsFromOem(path string) types.FileContents {
+	return types.FileContents{
+		Source: types.Url{
+			Scheme: "oem",
+			Path:   path,
+		},
+	}
+}
+
+func userCloudInit(name string, oem string) types.SystemdUnit {
+	return types.SystemdUnit{
+		Name:     "oem-cloudinit.service",
+		Enable:   true,
+		Contents: fmt.Sprintf("[Unit]\nDescription=Cloudinit from %s metadata\n\n[Service]\nType=oneshot\nExecStart=/usr/bin/coreos-cloudinit --oem=%s\n\n[Install]\nWantedBy=multi-user.target", name, oem),
```

Comment:
This format string is long enough that it'd be better as a multi-line raw string constant